flink-connectors/flink-connector-hbase/src/main/java/org/apache/flink/connector/hbase/source/AbstractTableInputFormat.java

```diff
@@ -186,24 +193,22 @@ public void close() throws IOException {
 			if (resultScanner != null) {
 				resultScanner.close();
 			}
-		} finally {
-			resultScanner = null;
-		}
-	}
-
-	@Override
-	public void closeInputFormat() throws IOException {
-		try {
 			if (table != null) {
 				table.close();
 			}
+			if (connection != null) {
+				connection.close();
+			}
 		} finally {
+			resultScanner = null;
 			table = null;
+			connection = null;
 		}
 	}
 
 	@Override
 	public TableInputSplit[] createInputSplits(final int minNumSplits) throws IOException {
+		initTable();
```

Comment:
This is never closed ?

Reply:
> This is never closed ?

This is closed when calling `close` method. I thought that in the lifecycle, `createInputSplits` would be called before `close`. Therefore, I didn't close the connection and table.

Reply:
I think it is called on the JobMaster [1], so we should use a local connection and close the connection in the method. 

[1]: https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionJobVertex.java#L258

Reply:
@wuchong I thought that table isn't close in this method, therefore the connection should not be closed in this method.